src/TextUI/Command.php

```diff
@@ -1118,9 +1143,11 @@ protected function showHelp(): void
   --printer <printer>         TestListener implementation to use
 
   --resolve-dependencies      Resolve dependencies between tests
+  --defects-first-order       Run previously unsuccessful tests first
   --random-order              Run tests in random order
   --random-order-seed=<N>     Use a specific random seed <N> for random order
   --reverse-order             Run tests last-to-first
```

Comment:
what's the outcome of setting up both, `--reverse-order` and `--defects-first-order` ? will it be respected, ignored, crashed ?

Reply:
looks like `--defects-first-order` overrides `--random-order` and `--reverse-order`

Reply:
I actually combines them. Each of the sorters has a specific preference and some are more important than others:
1. _random_ and _reverse_ go first, they are considered mostly cosmetic. In theory they have no effect on the outcome.
2. More important is sorting the defects to the front. It is an explicit request from the (human) user that indicates "I have seen what happens during the runs and I would like an explicit change to that."
3. The dependency resolver has the last say since it's job is preventing breakages. A side effect of _defects first_ I found is getting unrelated failures as the defects were sometimes sorted the wrong 'side' of their dependants.

Reply:
I agree I need to explicitly document this behaviour and am working on a few diagrams that show the limits of the sorters when looking at how suites can be inside other suites, what dependencies really do, and so on.

Reply:
if you need big description and diagrams, it's probably means it's overcomplicated.
maybe let us shrink it down to what would be the most value for user?

I would suggest sth like:
`--order-by=default|reverse|random|defect`
without possibility to mix them, like `reverse + defect`. I don't see much value of that combination, do you ?

Reply:
Yes, I think that would be better. My personal use case of 'randomized but with previous defects first' might be too exotic. Let's keep it simple.

Reply:
@sebastianbergmann do you have a preference. Guessing from your other comments, you prefer the simpler UX as well?